lib/sles4sap_publiccloud.pm

```diff
@@ -531,20 +533,40 @@ sub check_takeover {
 
 sub enable_replication {
     my ($self, %args) = @_;
-    croak("Argument <site_name> missing") unless $args{site_name};
+    croak('enable_replication [ERROR] Argument <site_name> missing') unless $args{site_name};
     my $hostname = $self->{my_instance}->{instance_id};
-    die("Database on the fenced node '$hostname' is not offline") if ($self->is_hana_database_online);
-    die("System replication '$hostname' is not offline") if ($self->is_primary_node_online);
+    my $remote_host;
+    my $sr_mode;
+    my $op_mode;
+    my $instance_id;
+    die("enable_replication [ERROR] Database on the fenced node '$hostname' is not offline") if ($self->is_hana_database_online);
+    die("enable_replication [ERROR] System replication '$hostname' is not offline") if ($self->is_primary_node_online);
 
     my $topology = $self->get_hana_topology();
-    foreach (qw(vhost remoteHost srmode op_mode)) { die "Missing '$_' field in topology output" unless defined(%$topology{$hostname}->{$_}); }
+    for my $site (keys %{$topology->{'Site'}}) {
+        foreach (qw(srMode opMode)) { die("enable_replication [ERROR] Missing '$_' field in topology output of Site->$site") unless defined($topology->{'Site'}->{$site}->{$_}); }
```

Comment:
can we split it in multiple line (just different indentation)

Reply:
Is this test on `srMode` and `opMode` to be performed here for all site even if we only care about `$args{site_name}` ?